core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -283,6 +283,9 @@ class ReplicaManager(val config: KafkaConfig,
   newGauge("AtMinIsrPartitionCount", () => leaderPartitionsIterator.count(_.isAtMinIsr))
   newGauge("OneAboveMinIsrPartitionCount", () => leaderPartitionsIterator.count(_.isOneAboveMinIsr))
   newGauge("ReassigningPartitions", () => reassigningPartitionsCount)
+  Partition.ISR_STATES_TO_CREATE_METRICS.foreach(c =>
```

Comment:
I think this metric would be helpful in investigating long latency tails for Produce request acks=all. We could use it to track how many partitions are still under PendingShrinking. (The leader brokers failing to shrink ISR to kick out the slow follower broker are the one experiencing long latency)